### PR TITLE
 armv7m: Update memory maps configuration to enable MPU 

### PR DESCRIPTION
**Pull request by agkaminski**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (refactoring, style fixes, git/CI config, submodule management, no code logic changes)

## How Has This Been Tested?

- [x] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: _targets/build.project.armv7m7-imxrt105x [hunk 1]
```diff
@@ -55,13 +55,15 @@ KERNEL_PHBASE=$((0x0)) # itcm start address
...
 
 # Pre-init script is launched before user script
 PREINIT_SCRIPT=(
 	"map itcm 0x0 0x28000 rwx"
 	"map dtcm 0x20000000 0x20048000 rw"
-	"map ocram 0x20200000 0x20220000 rwxc"
-	"map xip1 0x60000000 0x60800000 rx"
+	"map ocram 0x20200000 0x20220000 rwxcb"
+	"map xip1 0x60000000 0x60800000 rxcb"
+	"map io 0x40000000 0x50000000 rx"
```
Comment (lukileczo) [inline _targets/build.project.armv7m7-imxrt105x:66]: Shouldn't it be `rw`?
```suggestion
	"map io 0x40000000 0x50000000 rw"
```

File: _targets/build.project.armv7m7-imxrt117x [hunk 2]
```diff
@@ -70,8 +74,8 @@ PREINIT_SCRIPT=(
 USER_SCRIPT=(
 	"kernel ${BOOT_DEVICE}"
 	"app ${BOOT_DEVICE} -x dummyfs xip1 ocram2"
-	"app ${BOOT_DEVICE} -x imxrt-multi xip1 ocram2"
-	"app ${BOOT_DEVICE} -x psh xip1 ocram2"
+	"app ${BOOT_DEVICE} -x imxrt-multi xip1 ocram2;aips14;aips5"
+	"app ${BOOT_DEVICE} -x psh xip1 ocram2;flexspi"
```
Comment (gerard5) [inline _targets/build.project.armv7m7-imxrt117x:78]: Why `psh` needs FlexSPI TX/RX FIFO memory range for data map ?

File: _targets/build.project.armv7m7-imxrt117x [hunk 3]
```diff
@@ -54,13 +54,17 @@ KERNEL_PHBASE=$((0x0)) # itcm start address
...
-	"map dtcm 0x20000000 0x20028000 rwx"
-	"map ocram2 0x202c0000 0x20340000 rwxc"
-	"map xip1 0x30000000 0x30400000 rx"
+	"map itcm 0x0 0x40000 rwxcb"
+	"map dtcm 0x20000000 0x20028000 rwcb"
+	"map ocram2 0x202c0000 0x20340000 rwxcb"
+	"map aips14 0x40000000 0x41000000 rws"
+	"map aips5 0x42000000 0x42100000 rws"
+	"map xip1 0x30000000 0x30400000 rxcb"
+	"map flexspi 0x7f000000 0x80000000 rws"
```
Comment (gerard5) [inline _targets/build.project.armv7m7-imxrt117x:67]: I think we don't use this range directly, also `imxrt-flash` accesses flexspi through `0x400CC000 ..+0x460` (FSPI1), `0x400D0000 ..+0x460` (FSPI2) and this is `AIPS-4` range (imxrt117x case)

File: _targets/build.project.armv7m7-imxrt106x [hunk 4]
```diff
@@ -77,9 +78,9 @@ PREINIT_SCRIPT=(
 USER_SCRIPT=(
 	"kernel ${BOOT_DEVICE}"
 	"app ${BOOT_DEVICE} -x dummyfs xip1 ocram2"
-	"app ${BOOT_DEVICE} -x imxrt-multi xip1 ocram2"
+	"app ${BOOT_DEVICE} -x imxrt-multi xip1 ocram2;aips14;aips5"
 	"app ${BOOT_DEVICE} -x psh xip1 ocram2"
-	"app ${BOOT_DEVICE} imxrt-flash ocram2 ocram2"
+	"app ${BOOT_DEVICE} imxrt-flash ocram2 ocram2;flexspi"
```
Comment (gerard5) [inline _targets/build.project.armv7m7-imxrt106x:83]: `imxrt-flash` uses `0x402a8000..+0x300` (FSPI1)  and `0x402a4000..+0x300` (FSPI2) to access registers and TX/RX FIFO which is `AIPS-3` range, access through `0x7f000000..0x7fffffff` is not used directly.

```suggestion
	"app ${BOOT_DEVICE} imxrt-flash ocram2 ocram2;aips14"
```
btw, I wonder how `imxrt-flash` started without access to its registers (see above), it should fail with memmgmt fault during access to it's registers.

File: _targets/build.project.armv7m7-imxrt117x (hunk 2 above)
Comment (agkaminski) [reply _targets/build.project.armv7m7-imxrt117x]: It does not, removed

File: _targets/build.project.armv7m7-imxrt117x (hunk 3 above)
Comment (agkaminski) [reply _targets/build.project.armv7m7-imxrt117x]: Removed

File: _targets/build.project.armv7m7-imxrt106x (hunk 4 above)
Comment (agkaminski) [reply _targets/build.project.armv7m7-imxrt106x]: Let's test it. Maybe it won't access anything during init when we xip from the memory?

Comment (agkaminski) [reply _targets/build.project.armv7m7-imxrt106x]: imxrt-`flash` seems to do nothing with the registers on imxrt1064. `imxrt`-multi crashed successfully when I took it's access the i/o away:
```
 Exception: 4 #MemMgtFault
   r0=00000558  r1=0000007f  r2=40184000  r3=2024e1bc
   r4=00000000  r5=2024dcf0  r6=70036ebc  r7=2024dcf0
   r8=2024dcf0  r9=2024cfa4 r10=00000000 r11=70036f4c
  r12=2024e1a2 psr=01000000  lr=7002e6ed  pc=7002db5c
  psp=2024fcd0 msp=2000b800 exr=ffffffed bfa=4018402c
  cfs=00000082 mma=4018402c
```

Comment (agkaminski) [reply _targets/build.project.armv7m7-imxrt106x]: I changed `flexspi` to `aips14` as advised

Comment (gerard5) [reply _targets/build.project.armv7m7-imxrt106x]: `app ${BOOT_DEVICE} imxrt-flash ..` is without `-x`, that's why

Comment (agkaminski) [reply _targets/build.project.armv7m7-imxrt106x]: Haha, that explains it. I'll enable it

Comment (agkaminski) [reply _targets/build.project.armv7m7-imxrt106x]: Well, that's why it was disabled I guess:
```
 phoenix-rtos-devices/storage/imxrt-flash/flashsrv.c:871 error: imxrt-flashsrv: create_dev /dev/flash0 - err: -17
  phoenix-rtos-devices/storage/imxrt-flash/flashsrv.c:997 error: imxrt-flashsrv: flash memories were not initialized correctly.
```

Comment (agkaminski) [reply _targets/build.project.armv7m7-imxrt106x]: Not is scope of this PR

Comment (gerard5) [reply _targets/build.project.armv7m7-imxrt106x]: Yep, right. Looks like missing partition in the last sector.

File: _targets/build.project.armv7m7-imxrt105x [hunk 5]
```diff
@@ -55,13 +55,15 @@ KERNEL_PHBASE=$((0x0)) # itcm start address
 KERNEL_PHOFFS=$((0x0))
 export KERNEL_PHADDR=$(printf "%08x" $((KERNEL_PHBASE + KERNEL_PHOFFS)))
 
+# All data/code maps need to be cacheable to allow unaligned accesses
 
 # Pre-init script is launched before user script
 PREINIT_SCRIPT=(
 	"map itcm 0x0 0x28000 rwx"
```
Comment (Darchiv) [inline _targets/build.project.armv7m7-imxrt105x:62]: Shouldn't this be `rwxcb`?
```suggestion
	"map itcm 0x0 0x28000 rwxcb"
```

File: _targets/build.project.armv7m7-imxrt105x (hunk 1 above)
Comment (Darchiv) [reply _targets/build.project.armv7m7-imxrt105x:66]: Is this really fixed?

Comment (agkaminski) [reply _targets/build.project.armv7m7-imxrt105x]: Somehow I forgot/did not stage changes, it's fixed now

File: _targets/build.project.armv7m7-imxrt105x (hunk 5 above)
Comment (agkaminski) [reply _targets/build.project.armv7m7-imxrt105x]: From my testing it doesn't seem to do anything anyway, but you're right, will remove these attributes from tcms

Comment (agkaminski) [reply _targets/build.project.armv7m7-imxrt105x:62]: Hmm, on the other hand, if I remember correctly, it's needed anyway for unaligned access to work. I'll run the CI one last time

Comment (agkaminski) [reply _targets/build.project.armv7m7-imxrt105x:62]: Yup, it crashes, I'm restoring cacheable flags

Comment (agkaminski) [reply _targets/build.project.armv7m7-imxrt105x]: @gerard5 these flags do not change anything about behaviour of TCM, but if they're absent, unaligned access with cause usage fault (unaligned access is disallowed on strongly ordered memory, regardless of SCB config, it's specified int the cortex-m manual)

**Review by Darchiv — APPROVED** [review]

**Pull request merged**